zenoh/src/sample.rs

```diff
@@ -527,12 +527,26 @@ pub struct QoS {
     pub express: bool,
 }
 
-impl From<QoSType> for QoS {
-    fn from(ext: QoSType) -> Self {
-        QoS {
-            priority: ext.get_priority().into(),
+impl QoS {
+    /// Helper function to fallback to default QoS value and log a error in case of conversion failure
+    pub(crate) fn from_or_default(ext: QoSType) -> QoS {
```

Comment:
The only faillable conversion is `Priority`.
With current implementation the whole `QoS` is falling back to default while only the `Priority` field should be resetted to default. `congestion_control` and `express` fields shouldn't be changed.

Reply:
fixed